spec/Naming.md

```diff
@@ -0,0 +1,122 @@
+* Namespace = scheme:[//authority] (the datasource)
+* Name = path (the datasets)
+
+### Data warehouses/data bases.
+Datasets are called tables. Tables are organised in databases and schemas.
+#### Postgres:
+Datasource hierarchy:
+ * Host
+ * Port
+
+Naming hierarchy:
+ * Database
+ * Schema
+ * Table
+
+Identifier:
+ * Namespace: postgres://{host}:{port} of the service instance.
+   * Scheme = postgres
+   * Authority = {host}:{port}
+ * Unique name: {database}.{schema}.{table}
+   * URI =  postgres://{host}:{port}/{database}.{schema}.{table}
+  
+#### Snowflake
+See: [Object Identifiers — Snowflake Documentation](https://docs.snowflake.com/en/sql-reference/identifiers.html)
+
+Datasource hierarchy:
+ * account name
+
+Naming hierarchy:
+ * Database: {database name} => unique across the account
+ * Schema: {schema name} => unique within the database
+ * Table: {table name} => unique within the schema
+
+Identifier:
+ * Namespace: snowflake://{account name}
+   * Scheme = snowflake
+   * Authority = {account name}
+ * Name: {database}.{schema}.{table}
+   * URI =   snowflake://{account name}/{database}.{schema}.{table}
+
+#### BigQuery
+See: 
+[Creating and managing projects | Resource Manager Documentation](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
+[Introduction to datasets | BigQuery](https://cloud.google.com/bigquery/docs/datasets-intro)
+[Introduction to tables | BigQuery](https://cloud.google.com/bigquery/docs/tables-intro)
+
+Datasource hierarchy:
+ * bigquery
+
+Naming hierarchy:
+ * Project Name: {project name} => is not unique
+ * Project number: {project number} => numeric: is unique across google cloud
+ * Project ID: {project id} => readable: is unique across google cloud
+ * dataset: {dataset name} => is unique within a project
+ * table: {table name} => is unique within a dataset
+
+Identifier :
+ * Namespace: bigquery
+   * Scheme = bigquery
+   * Authority = 
+ * Unique name: {project id}.{dataset name}.{table name}
+   * URI =   bigquery:{database}.{schema}.{table}
+
+### Distributed file systems/blob stores
+#### GCS
+Datasource hierarchy: none, naming is global
+
+Naming hierarchy:
+ * bucket name => globally unique
+ * Path 
+
+Identifier :
+ * Namespace: gs://{bucket name}
+   * Scheme = gs
+   * Authority = {bucket name}
+ * Unique name: {path}
+   * URI =   gs://{bucket name}{path}
+
+#### S3
+Naming hierarchy:
+ * bucket name => globally unique
+ * Path 
+
+Identifier :
+ * Namespace: s3://{bucket name}
+   * Scheme = s3
+   * Authority = {bucket name}
+ * Unique name: {path}
+   * URI =   s3://{bucket name}{path}
+
+#### HDFS
+Naming hierarchy:
+ * Namenode: host + port
+ * Path
+
+Identifier :
+ * Namespace: hdfs://{namenode host}:{namenode port}
+   * Scheme = hdfs
+   * Authority = {namenode host}:{namenode port}
+ * Unique name: {path}
+   * URI =   hdfs://{namenode host}:{namenode port}{path}
+
+### Hive
+
```

Comment:
I assume here, something like

> Naming hierarchy:
>   * Metastore host
>   * Database name
>   * Table name
>
> Identifier :
>  * Namespace: hive://{metastore host}:{metastore port}
>    * Scheme = hive
>    * Authority = {metastore host}:{metastore port}
>  * Unique name: {database name}.{table name}
>    * URI =  hive://{metastore host}:{metastore port}/{database name}.{table name}

Reply:
Ah yes, this was a TODO. Item, should I leave it out for now or add this to the spec?

Reply:
Seems to me that it would be just as easy to add it in now vs later.

Reply:
The others have been researched and there are existing integrations. This is not the case yet for Hive. So I'd rather add it later.